docs/building-the-candidate-app/data-api.md

```diff
@@ -0,0 +1,42 @@
+# Building the data API between the frontend and Strapi or other backend solution
+
+A design principle of the OpenVAA framework is to make its functionalities (or modules) as independent of each other as possible. For connecting the SvelteKit frontend to the Strapi backend, this means we want to abstract the interface between the two so that any read and write calls from the frontend use functions and data types that are agnostic of the backend.
+
+The relevant TS interface is not yet defined for the Voter App, but it uses the API calls below, defined in [`/src/lib/api/getData.ts`](../../frontend/src/lib/api/getData.ts). The functions the Candidate App uses for making further read calls and write calls should follow a similar logic, where any specifics to Strapi are handled inside the functions and their outputs are properties that can be consumed by the frontend.
+
+```ts
```

Comment:
This is kind of good, but I guess we will struggle to maintain the examples. Maybe it is better to give a more abstract explanation. Something like, We have methods used to fetch data for each one of the Database tables or something similar

Reply:
I'll just leave one example.